.github/workflows/ci.yaml

```diff
@@ -91,11 +91,15 @@ jobs:
       - name: build dependencies
         run: stack --no-terminal build --fast --only-dependencies
       - name: build
-        run: stack --no-terminal build --fast
+        run: stack --no-terminal build --fast --no-run-tests --test
```

Comment:
What does this do, build tests but not run them?

Reply:
Yeah; without it, on the `stack test` line(s) below, stack will stupidly rebuild _library_ components of packages with test suites (and their dependencies, and so on).

It's a stack caching bug caused by stack treating every component of a package as one atomic unit.

Reply:
We didn't have it before because we only had one test suite, which was an executable component. Now, there are a couple packages with actual test suites that are run with `stack test`.